### PR TITLE
TAS-103/Ajouter_Credit_Card

### PR DESCRIPTION
**Pull request by Brendan-bx**

### CODE REVIEW

File: apps/api/src/routes/users/creditCards.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { Hono } from "hono"
+import { PrivateContextVariables } from "@alittlebyte/api/utils/types"
+
+export const creditCardRouter = () =>
+	new Hono<{ Variables: PrivateContextVariables }>()
+		.get("/", (c) => {
+			return c.json({})
+		})
```
Comment (Limerio) [inline apps/api/src/routes/users/creditCards.ts]: remove those lines

File: apps/api/src/routes/users/creditCards.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+		.post("/", async (c) => {
+			const { prisma } = c.var
+			const data = await c.req.json()
+			try {
+				const newCreditCard = await prisma.creditCard.create({ data })
+				return c.json(newCreditCard, 201)
```
Comment (Limerio) [inline apps/api/src/routes/users/creditCards.ts]: There is a problem with your query.
I'm not sure if the user exists or not. Use the param `:userId` to check if the user exists

**Review by Limerio — CHANGES_REQUESTED** [review]

**Review by WookiesRpeople2 — DISMISSED** [review]

File: apps/api/src/routes/users.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Hono } from "hono"
+import { PrivateContextVariables } from "@alittlebyte/api/utils/types"
+import { creditCardRouter } from "@alittlebyte/api/routes/users/creditCards"
+
+export const usersRouter = () =>
+	new Hono<{ Variables: PrivateContextVariables }>().route(
+		"/:userId/credit-cards",
+		creditCardRouter(),
```
Comment (Limerio) [inline apps/api/src/routes/users.ts]: Add a middleware to check if the connected user id is equal to the userId param 
```suggestion
                isLoginUser,
		creditCardRouter(),
```

File: apps/admin/package.json [hunk 4]
```diff
@@ -15,6 +15,7 @@
 		"@alittlebyte/components": "workspace:*",
 		"@tanstack/react-query": "^5.56.2",
 		"@tanstack/react-router": "^1.58.9",
+		"autoprefixer": "^10.4.20",
```
Comment (Limerio) [inline apps/admin/package.json]: remove this line
```suggestion
```

File: apps/api/src/middlewares/isUserEqualtoUserLogin.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+	{
+		req,
+		var: { user },
+		json,
+	}: Context<{ Variables: PrivateContextVariables }, string, UserIdInput>,
+	next: Next,
+) => {
+	const { userId } = req.valid("param")
+	if (userId !== user.id) {
+		return json({ error: "Forbidden" }, 403)
```
Comment (Limerio) [inline apps/api/src/middlewares/isUserEqualtoUserLogin.ts]: magic number 🌟
take a look at the `HTTP_STATUS_CODES` and use one of these values

File: apps/api/src/routes/users/creditCards.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+			const { prisma } = c.var
+			const data = c.req.valid("json")
+			try {
+				const newCreditCard = await prisma.creditCard.create({
+					data: {
+						...data,
+						userId,
+					},
+				})
+				return c.json(newCreditCard, 201)
```
Comment (Limerio) [inline apps/api/src/routes/users/creditCards.ts]: magic number 🌟
take a look at the `HTTP_STATUS_CODES` and use one of these values

File: apps/api/src/routes/users/creditCards.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+			} catch (error) {
+				console.error(error)
+				return c.json({ error: "Failed to create credit card" }, 500)
```
Comment (Limerio) [inline apps/api/src/routes/users/creditCards.ts]: magic number 🌟
take a look at the `HTTP_STATUS_CODES` and use one of these values

File: apps/landing/vite.config.ts.timestamp-1729632290581-391b06fb3d405.mjs
Comment (Limerio) [inline apps/landing/vite.config.ts.timestamp-1729632290581-391b06fb3d405.mjs:1]: remove this file

File: packages/common/src/validators.ts [hunk 8]
```diff
@@ -1,4 +1,16 @@
...
+export const idValidator = z
+	.string()
+	.uuid()
+	.transform((arg) => arg as UUID)
+
+export const cardNumberValidator = z.string().min(16).max(16)
+
+export const expirationDateValidator = z.date()
+
+export const cvvCodeValidator = z.string().min(3).max(3)
```
Comment (Limerio) [inline packages/common/src/validators.ts]: moove it on the bottom on the file

**Review by Limerio — CHANGES_REQUESTED** [review]

**Review by Limerio — APPROVED** [review]

**Pull request merged**